USER_GUIDE.md

```diff
@@ -104,9 +110,12 @@ func main() {
 
 	searchResponse, err := search.Do(context.Background(), client)
 	if err != nil {
-		fmt.Println("failed to search document ", err)
+		fmt.Println(fmt.Sprintf("failed to search document: %s", err))
 		os.Exit(1)
 	}
+	if err := searchResponse.Err(); err != nil {
+		fmt.Println(fmt.Sprintf("failed to search document: %s", err))
+	}
```

Comment:
I think this is net worse than what we had - we now have to check for error twice? Shouldn't `err` the first time be what we put into `searchResponse.Err()`?

Reply:
That's what I am saying, shouldn't we make `.Do` care about the actual response?

As a user I want to write this:

```go
createIndexResponse, err := createIndex.Do(context.Background(), client)
if err != nil {
  // err here is either a network error, or the error from the response
  fmt.Println(fmt.Sprintf("failed to create index: %s", err))
  os.Exit(1)
}
```

WDYT?

Reply:
Depends on what the lib maintainers wants.
Pro:
- The lib has a standard way on handling response errors, which prevent other users from doing it by them self
- You only need to check one error
- Adds the possibility to have a custom error type for some requests

Cons:
- Existing users with their own implementation needs to check the returned error. For example if the status code 404 is accepted. Currently this would be an error. Otherwise the question turns up, if the accepted response code should be a config setting to each Do request?

Reply:
https://github.com/opensearch-project/opensearch-go/compare/main...Jakob3xD:opensearch-go:opensearchapi-response-error

Still missing some stuff but this is how it could look like.

Reply:
I like the pros and I like the implementation. The fact that it's easier for beginners is huge.

Is this (the con) a breaking change for existing users? If so we'll need to major increment the version of the client.